### ISSUE TITLE
Interpreting summary results

### ISSUE THREAD

**Issue by micdic** [mahmoodlab/CLAM#94]
Hi, i'm testing this really interesting framework on a subtyping problem (3 classes).

I've just filtered my dataset to avoi blurred images 
and i started training the model with a balanced dataset with about 50 WSI per class. 

I've attached here some summary results, but i'm facing some problem in their interpretation.
It happens that in some cases a good AUC is not related to a good accuracy and vice versa.
In the paper you mentioned that you take the average AUC of models in your experiments.
What is in your experience a "good" behaviour in training (due to the attention_net + clustering approach for the top patches),
looking at the summary results? I see very different values generally, not having let's say "homogenous" results (poor or good) as expected.

As an example, if you look at this summary table reported here:
---------------------------------
,folds,test_auc,val_auc,test_acc,val_acc
,  folds,  test_auc,  val_auc,  test_acc,  val_acc
0,  0,  0.5729166666666666,  0.7428774928774929,  0.16666666666666663,  0.5882352941176471
1,  1,   0.8089799556186111,  0.8750240500240499,  0.7142857142857143,  0.6875
2,  2,  0.8165584415584416,  0.9142857142857143,  0.6666666666666667,  0.8333333333333334
3,  3,  0.8203703703703704,  0.9458333333333334,  0.6923076923076923,  0.7142857142857143
4,  4,  0.7152176527176527,  0.9707936507936509,  0.5,  0.8
5,  5,  0.8432772166105499,  0.8286324786324787,  0.5333333333333333,  0.7058823529411764
6,  6,  0.6583333333333333,  0.9125,  0.4285714285714286,  0.8571428571428572
7,  7,   0.8637566137566138,  0.8439842873176207,  0.6923076923076923,  0.7333333333333334
8,  8,  0.8833333333333333,  0.8666666666666667,  0.7222222222222222,  0.6666666666666667
9,  9,  0.6153439153439154,  0.7207120811287479,  0.2727272727272727,  0.5
----------------------------------
results seem to be some how (at least in some cases) inconsistent, i.e. val_acc > test_acc (folder 2).
Of course overfitting is around the corner.. ;)
Do you have some hints about?, did you experiment some "outlier results" in your test or a much more "linear" performance?
Thanks in advance for you time.
Michele.

P.s.
In some cases i've also  found some strange integer values in summary columns, like 625....

**Comment by fedshyvana** [comment]
Can you describe your dataset a bit more? is the 150 WSIs (50 per class) in total, that then gets split into train/val/test? How big is your val and test set then? If you e.g. set aside 10% data for testing, that would be n = 15 in your test - which is extremely small, and might sometimes not be representative of your actual data distribution at all - hence the high variance you see from fold to fold.

**Comment by micdic** [comment]
Thanks for your fast reply. 

Yes, the dataset (totally) is made of about 150 WSI (50 per class), so the splitting reduced the dataset.

Results of create_splits_seq-py
---------------------------------------
number of classes: 3
slide-level counts:  
0    48
1    51
2    50
Name: label, dtype: int64
Patient-LVL; Number of samples registered in class 0: 21
Slide-LVL;   Number of samples registered in class 0: 48
Patient-LVL; Number of samples registered in class 1: 29
Slide-LVL;   Number of samples registered in class 1: 51
Patient-LVL; Number of samples registered in class 2: 21
Slide-LVL;   Number of samples registered in class 2: 50
num_slides_cls [21 29 21]
args.val_frac 0.1
val_num [2 3 2]
args.test_frac 0.1
test_num [2 3 2]
label_fracs [1.0]
val_num [2 3 2]
----------------------------------------------------------
Effectively an "average" splitting scenario obtained is like this:

number of training samples: 122
number of samples in cls 0: 41
number of samples in cls 1: 39
number of samples in cls 2: 42

number of val samples: 12
number of samples in cls 0: 3
number of samples in cls 1: 5
number of samples in cls 2: 4

number of test samples: 15
number of samples in cls 0: 4
number of samples in cls 1: 7
number of samples in cls 2: 4

Too few samples in val/test evaluation....
I will try to enlarge dataset.

One more insight, the evaluation of only B patches (the highest/lowest) out form the attention_net, is some how related, in your architetcure, to the total number of available patches? I mean, do you estimate the threshold B considering the whole number of patches? Or there is some other "rules" in selecting the value of B? 

Sorry for posting other questions and thanks for your time.

**Comment by fedshyvana** [comment]
You can treat B like a hyperparamter, in the paper we tested like B from {8,32,64,128) i remember 32 seem to work pretty well for most cases.

**Comment by fedshyvana** [comment]
feel free to follow-up if you have other issues
